### DISCUSSION TITLE
Is it possible to view site pictures?

### DISCUSSION THREAD

**Discussion by ltsz16** [justcoding121/Titanium-Web-Proxy#906] [Q&A] [answered]
Is it possible to view all images on a page in the proxy without having to parse through the website code for IMG tages and then download them separately and then view them?  We are looking at tieing in an image filter as part of the proxy.

**Comment by honfika** [comment, marked as answer]
If the image is downloaded trough the proxy, then yes, it is possible.

**Comment by justcoding121** [comment]
Yes ? if the browser loads images through the proxy, you'll see them as normal sessions. No need to scrape `<img>` tags yourself.

```csharp
proxyServer.BeforeResponse += async (sender, e) =>
{
    var contentType = e.HttpClient.Response.ContentType;
    if (contentType == null || !contentType.StartsWith("image/", StringComparison.OrdinalIgnoreCase))
        return;

    var bytes = await e.GetResponseBody();
    // run your image filter / save / block here
    // e.SetResponseBody(modifiedBytes); // if you alter it
    // or e.Ok(...) / custom 403 if you want to block
};
```

Also useful: in `BeforeRequest` / `BeforeTunnelConnectRequest` you can skip decrypt for hosts you don't care about so you only inspect image traffic you need.

Closing this out.
